### PR TITLE
Solution bug Problem custom name sequeceGenerator YAML why not create…

### PR DESCRIPTION
**Pull request by monstrodopantano**
Hi please accept my pull request to fixed custom sequence name. [#5789](https://github.com/doctrine/doctrine2/issues/5789)
best regards

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ lib/Doctrine/DBAL
 .idea
 vendor/
 composer.lock
+nbproject/
```
Comment (Ocramius) [inline .gitignore]: Please use your [global gitignore file](https://help.github.com/articles/ignoring-files/)

File: lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php [hunk 2]
```diff
@@ -655,14 +655,15 @@ private function completeIdGeneratorMapping(ClassMetadataInfo $class)
             case ClassMetadata::GENERATOR_TYPE_SEQUENCE:
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php]: Please do not change file FS flags

File: lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php [hunk 3]
```diff
@@ -317,6 +317,10 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
                     if (isset($fieldMapping['generator']['strategy'])) {
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Please do not change file FS flags

File: lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php [hunk 4]
```diff
@@ -655,14 +655,15 @@ private function completeIdGeneratorMapping(ClassMetadataInfo $class)
...
                 $definition = $class->sequenceGeneratorDefinition;
-
-                if ( ! $definition) {
+                $customDefinition = $class->customGeneratorDefinition;
+                if ( ! $definition || is_array($customDefinition) ) {
                     $fieldName      = $class->getSingleIdentifierFieldName();
                     $sequenceName   = $class->getSequenceName($this->getTargetPlatform());
                     $quoted         = isset($class->fieldMappings[$fieldName]['quoted']) || isset($class->table['quoted']);
 
+                    $schemaElementName = is_array($customDefinition) ? $customDefinition['sequenceName'] : $this->getTargetPlatform()->fixSchemaElementName($sequenceName);
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php]: This check seems to be duplicated

File: lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php [hunk 5]
```diff
@@ -317,6 +317,10 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
...
                         $metadata->setIdGeneratorType(constant('Doctrine\ORM\Mapping\ClassMetadata::GENERATOR_TYPE_'
                             . strtoupper($fieldMapping['generator']['strategy'])));
+                        if(isset($fieldMapping['generator']['sequenceGenerator'])
+                            && strtoupper($fieldMapping['generator']['strategy']) == 'SEQUENCE'){
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Please use `===` when comparing

Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Also check CS: Space before `{`

File: lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php [hunk 6]
```diff
@@ -317,6 +317,10 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
                     if (isset($fieldMapping['generator']['strategy'])) {
                         $metadata->setIdGeneratorType(constant('Doctrine\ORM\Mapping\ClassMetadata::GENERATOR_TYPE_'
                             . strtoupper($fieldMapping['generator']['strategy'])));
+                        if(isset($fieldMapping['generator']['sequenceGenerator'])
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Check CS: space after `if`

Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: This block needs to be tested (unit or integration test needed)

File: lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php [hunk 7]
```diff
@@ -655,14 +655,15 @@ private function completeIdGeneratorMapping(ClassMetadataInfo $class)
             case ClassMetadata::GENERATOR_TYPE_SEQUENCE:
                 // If there is no sequence definition yet, create a default definition
                 $definition = $class->sequenceGeneratorDefinition;
-
-                if ( ! $definition) {
+                $customDefinition = $class->customGeneratorDefinition;
+                if ( ! $definition || is_array($customDefinition) ) {
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php]: This added scenario needs tests

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 8]
```diff
@@ -11,4 +11,4 @@ lib/Doctrine/DBAL
 .project
```
Comment (Ocramius) [inline .gitignore]: Please don't change FS flags here either

**Comment by monstrodopantano** [comment]
Hi Ocraminus I fixed all bug requested. Only test scenario I will not do it due complexity framework. If you can not do the test then ignore my correction for bug.
best regards

**Comment by Ocramius** [comment]
@monstrodopantano please look into https://github.com/doctrine/doctrine2/tree/d3f6c5ec70aac4b029a4b61ecf1e2ba61a1a4a6d/tests/Doctrine/Tests/ORM/Functional/Ticket for example tests. Without a test, a patch is never going to be considered.

**Comment by ostrolucky** [comment]
I'm sorry, but this issue is no longer relevant in Doctrine 3.0

**Pull request closed without merging**